### PR TITLE
[4.0] Form Field Standardization

### PR DESCRIPTION
**Pull request by James-Antrim**
Pull Request for Issue # .

### Summary of Changes

Removed the unused attribute $inputType possible duplicate of $options.
Removed the inappropriate attribute $accept from several files.
Removed the redundant and incorrect attribute $spellcheck from several files.
Alphabetized CheckboxField attributes.
Outsourced the CheckboxField HTML generation to a layout.
Removed redundant options setting from ComboField and TagField.

### CODE REVIEW

File: layouts/joomla/form/field/color/advanced.php [hunk 1]
```diff
@@ -39,7 +41,6 @@
  * @var   string   $validate        Validation rules to apply.
  * @var   string   $value           Value attribute of the field.
  * @var   array    $checkedOptions  Options that will be set as checked.
- * @var   boolean  $hasValue        Has this field a value assigned?
  * @var   array    $options         Options available for this field.
```
Comment (Quy) [inline layouts/joomla/form/field/color/advanced.php]:
```suggestion
```

File: layouts/joomla/form/field/color/simple.php [hunk 2]
```diff
@@ -41,7 +43,6 @@
  * @var   string   $validate        Validation rules to apply.
  * @var   string   $value           Value attribute of the field.
  * @var   array    $checkedOptions  Options that will be set as checked.
- * @var   boolean  $hasValue        Has this field a value assigned?
  * @var   array    $options         Options available for this field.
```
Comment (Quy) [inline layouts/joomla/form/field/color/simple.php]:
```suggestion
```

File: administrator/components/com_workflow/View/Stages/HtmlView.php [hunk 3]
```diff
@@ -159,9 +160,19 @@ protected function addToolbar()
 
 		$toolbar = Toolbar::getInstance('toolbar');
 
-		$workflow = !empty($this->state->get('active_workflow', '')) ? Text::_($this->state->get('active_workflow', '')) . ': ' : '';
+		$workflow = '';
+		if (!empty($this->workflowID)) {
```
Comment (Quy) [inline administrator/components/com_workflow/View/Stages/HtmlView.php]:
```suggestion

		if (!empty($this->workflowID))
		{
```

File: administrator/components/com_workflow/View/Stages/HtmlView.php [hunk 4]
```diff
@@ -159,9 +160,19 @@ protected function addToolbar()
...
```
Comment (Quy) [inline administrator/components/com_workflow/View/Stages/HtmlView.php:168]:
```suggestion
```

File: administrator/components/com_workflow/View/Transitions/HtmlView.php [hunk 5]
```diff
@@ -147,9 +148,19 @@ protected function addToolbar()
 
 		$toolbar = Toolbar::getInstance('toolbar');
 
-		$workflow = !empty($this->state->get('active_workflow', '')) ? Text::_($this->state->get('active_workflow', '')) . ': ' : '';
+		$workflow = '';
+		if (!empty($this->workflowID)) {
```
Comment (Quy) [inline administrator/components/com_workflow/View/Transitions/HtmlView.php]:
```suggestion

		if (!empty($this->workflowID))
		{
```

File: administrator/components/com_workflow/View/Transitions/HtmlView.php [hunk 6]
```diff
@@ -147,9 +148,19 @@ protected function addToolbar()
...
```
Comment (Quy) [inline administrator/components/com_workflow/View/Transitions/HtmlView.php:156]:
```suggestion
```

File: administrator/components/com_workflow/forms/filter_stages.xml [hunk 7]
```diff
@@ -7,6 +7,16 @@
 			label="COM_WORKFLOW_FILTER_SEARCH_LABEL"
 			hint="JSEARCH_FILTER"
 		/>
+        <field
```
Comment (Quy) [inline administrator/components/com_workflow/forms/filter_stages.xml]: Please use tabs and not spaces to indent.

File: administrator/language/en-GB/en-GB.com_workflow.ini [hunk 8]
```diff
@@ -21,7 +21,6 @@ COM_WORKFLOW_DATE_CREATED="Date Created"
 COM_WORKFLOW_DATE_MODIFIED="Date Modified"
 COM_WORKFLOW_DEFAULT="Default"
 COM_WORKFLOW_DEFAULT_ITEM="Default option is already set for a different item."
-COM_WORKFLOW_DEFAULT_WORKFLOW="Joomla! Default"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_workflow.ini]: Not sure I understand why you have removed this string and then hardcoded the english in the sql install

Comment (James-Antrim) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: It's back the way it was in the workflow branch for which the second commit was actually intended.

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: Sorry I dont understand you.

Comment (James-Antrim) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: https://github.com/bembelimen/joomla-cms/blob/workflow-extended/administrator/language/en-GB/en-GB.com_workflow.ini

Comment (James-Antrim) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: https://github.com/bembelimen/joomla-cms/blob/workflow-extended/installation/sql/mysql/joomla.sql

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: The problem then is that this PR will change that if it is merged second. If you do not want to change it then you should update it here in this PR.

Comment (James-Antrim) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: Done

File: installation/sql/postgresql/joomla.sql [hunk 9]
```diff
@@ -2266,7 +2266,7 @@ CREATE INDEX "#__workflows_idx_modified" ON "#__workflows" ("modified");
 CREATE INDEX "#__workflows_idx_modified_by" ON "#__workflows" ("modified_by");
 
 INSERT INTO "#__workflows" ("id", "asset_id", "published", "title", "description", "extension", "default", "core", "ordering", "created", "created_by", "modified", "modified_by") VALUES
-(1, 56, 1, 'COM_WORKFLOW_DEFAULT_WORKFLOW', '', 'com_content', 1, 1, 1, '1970-01-01 00:00:00', 0, '1970-01-01 00:00:00', 0);
+(1, 56, 1, 'Joomla! Default', '', 'com_content', 1, 1, 1, '1970-01-01 00:00:00', 0, '1970-01-01 00:00:00', 0);
```
Comment (infograf768) [inline installation/sql/postgresql/joomla.sql]: revert?

File: administrator/language/en-GB/en-GB.com_workflow.ini (hunk 8 above)
Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: Thanks but you missed one in installation/sql/postgresql/joomla.sql

**Comment by infograf768** [comment]
Test instructions?

File: administrator/components/com_workflow/View/Stages/HtmlView.php (hunk 4 above)
Comment (James-Antrim) [reply administrator/components/com_workflow/View/Stages/HtmlView.php:168]: This no would no longer set the text to be used for the workflow name output.

File: administrator/components/com_workflow/View/Transitions/HtmlView.php (hunk 6 above)
Comment (James-Antrim) [reply administrator/components/com_workflow/View/Transitions/HtmlView.php:156]: This no would no longer set the text to be used for the workflow name output.

File: installation/sql/postgresql/joomla.sql (hunk 9 above)
Comment (James-Antrim) [reply installation/sql/postgresql/joomla.sql]: Sorry missed that one. One moment.

Comment (James-Antrim) [reply installation/sql/postgresql/joomla.sql]: Done.

File: administrator/language/en-GB/en-GB.com_workflow.ini (hunk 8 above)
Comment (James-Antrim) [reply administrator/language/en-GB/en-GB.com_workflow.ini]: Sorry, done.

File: administrator/components/com_workflow/forms/filter_stages.xml (hunk 7 above)
Comment (James-Antrim) [reply administrator/components/com_workflow/forms/filter_stages.xml]: Done.

**Comment by SharkyKZ** [comment]
@James-Antrim There are some unrelated changes here, possibly due to conflicts. Are you going to update this PR?

**Comment by James-Antrim** [comment]
> @James-Antrim There are some unrelated changes here, possibly due to conflicts. Are you going to update this PR?

As far as I am concerned this is done.

The continuous integration errors have nothing to do with my code.

Someone is modifying the h1.page-title after it has already been rendered, giving it an id with the modified contents of the icon span. The title layout was last changed in 2017 by 'Walsh' and the mod_title default was last modified in 2018 by yourself. Upto and including the default.php for mod_title the output is as expected. What some js somewhere does after the fact is not my business.

**Comment by SharkyKZ** [comment]
Tests are probably unrelated. I'm referring to some Workflow-related changes you have in here. They seem to be added by mistake as they not related to "Form Field Standardization"?

**Comment by James-Antrim** [comment]
> Tests are probably unrelated. I'm referring to some Workflow-related changes you have in here. They seem to be added by mistake as they not related to "Form Field Standardization"?

Yes. I had mistakenly not opened a new branch for a workflow related commit. The changes to the workflow part have already been made a part of Benjamin's workflow fork, and my commits reverting my change from constant to text back to a constant were already done there, as I believe I wrote in my reply to Brian. At Brian's suggestion I reverted them in this as well as it had already made it into the 4.0 branch. The still unmerged changes should only be that revert, no new code per se.

**Comment by SharkyKZ** [comment]
But Workflow changes are still included in this PR.

**Comment by BertaOctech** [comment]
Could you include some testing instructions, please?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24786">issues.joomla.org/tracker/joomla-cms/24786</a>.</sub>

**Pull request closed without merging**